addons/survey/static/src/js/survey_form.js

```diff
@@ -1167,10 +1171,20 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
     // -------------------------------------------------------------------------
 
     _showErrors: function (errors) {
-        var self = this;
-        var errorKeys = _.keys(errors);
+        const self = this;
+        const errorKeys = _.keys(errors);
         _.each(errorKeys, function (key) {
-            self.$("#" + key + '>.o_survey_question_error').append($('<p>', {text: errors[key]})).addClass("slide_in");
+            let msg = errors[key];
+            // Matrix missing rows highlight
+            if (errors[key] instanceof Object){
+                msg = errors[key].msg;
+                errors[key].row_ids.forEach(row_id => {
+                    let row = document.querySelector("div[id='" + key + "']>.o_survey_question_matrix>tbody>tr[id='" + row_id + "']>th")
```

Comment:
For the query selectors, we could scope the search to the widget container by using `this.el` (i.e: `this.el.querySelector("...")`) That way, we can avoid side effects if we ever have the same widget several times on the page.